services/api/src/api/routes/endpoint.py

```diff
@@ -51,60 +156,31 @@ def create_endpoint(
     max_age_short: int = 0,
 ) -> Endpoint:
     async def processing_step_endpoint(request: Request) -> Response:
-        if not processing_steps:
-            raise MissingProcessingStepsError("Missing processing steps or empty list")
-
-        processing_step = processing_steps[0]
-        # TODO: Add support to iterate over the processing steps list in case of failure
         try:
-            dataset = request.query_params.get("dataset")
-            if not are_valid_parameters([dataset]) or not dataset:
-                raise MissingRequiredParameterError("Parameter 'dataset' is required")
-            if processing_step.input_type == "dataset":
-                config = None
-                split = None
-            elif processing_step.input_type == "config":
-                config = request.query_params.get("config")
-                split = None
-                if not are_valid_parameters([config]):
-                    raise MissingRequiredParameterError("Parameter 'config' is required")
-            else:
-                config = request.query_params.get("config")
-                split = request.query_params.get("split")
-                if not are_valid_parameters([config, split]):
-                    raise MissingRequiredParameterError("Parameters 'config' and 'split' are required")
-            logging.info(f"{processing_step.job_type}, dataset={dataset}, config={config}, split={split}")
+            # getting input params
+            params: InputParameters = get_input_parameters(request.query_params)
+            logging.info(
+                f"input_type={params.input_type}, dataset={params.dataset}, config={params.config},"
+                f" split={params.split}"
+            )
+
+            if params.input_type not in steps_by_input_type:
+                raise MissingRequiredParameterError("Operation not supported or not found")
```

Comment:
also, I think the pythonic way would be something like

```python
try:
  processing_steps = steps_by_input_type[params.input_type]
except KeyError as e:
  raise MissingRequiredParameterError("Operation not supported or not found") from e
```

instead of

```python
if params.input_type not in steps_by_input_type:
  raise MissingRequiredParameterError("Operation not supported or not found")

processing_steps = steps_by_input_type[params.input_type]
```

@albertvillanova is the pythonic advisor here :P

Reply:
> Is it what we want? (maybe)

The problem is that now that we support having different responses based on query params, we need to infer the input param to use.
What if the inferred input param is not supported?
I thought in these possible scenarios (Almost sure those are the unique ones):
- endpoint supporting only dataset input type
  - request with dataset param-> OK
  - request with dataset and config params-> Ignone the other fields
  - request with dataset, config and split params> Ignone the other fields
  - request with dataset, config, split and somethingelse params-> Ignone the other fields
- endpoint supporting dataset and config input types
  - request with dataset param-> Which of them should we use? dataset -> OK, config-> Throw missing param error
  - request with dataset and config params -> Which of them should we use? it will work for both, for dataset ignoring config and for config using both params 
  - request with dataset, config and split params-> Which of them should we use? It will work for dataset and config but for split-> Missing param error
  - request with dataset, config, split and somethingelse params -> Same as above, we could ignore the somethingelse param
- endpoint supporting dataset, config and split input types
  - request with dataset param-> Which input type should we use? dataset -> OK, config -> Missing param, split -> Missing param
  - request with dataset and config params -> Which input type should we use (dataset/config)? , split -> Missing param
  - request with dataset, config and split params -> Which input type should we use? it could work with all of them
  - request with dataset, config, split and somethingelse params -> Same as above, we could ignore the somethingelse param
  
  As you can see, there are many possibilities, I didn't want to complicate the validation.
  I thought the simple way to validate it could be only look for dataset,config and split param, if there is any other param they will be ignored (Like somethingelse=true ..).
  Based on dataset,config and split get the input type to use and if we dont support that logic, throw error.

What do you think? @huggingface/datasets-server

Reply:
I think I have a better idea, considering that each endpoint should support at least one associated input type:
1. Get the inferred input type from request params
2. Get the supported sorted input type list by the endpoint  (split > config > dataset , e.g if the endpoint supports all the input types, it will return [split, config, dataset], if the endpoint supports config and split, it will return [split,config])
3. Iterate over supported sorted input type list and if matches inferred input type, get the result based on its processing steps, else, store corresponding input type message
   -  split ("Parameters 'split', 'config' and 'dataset' are required")
   -  config ("Parameters 'config' and 'dataset' are required")
   -  dataset ("Parameter 'dataset' is required)
4.  At the end, if no one supported input type matched the inferred input type, return the last error message